dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java

```diff
@@ -1010,31 +1010,53 @@ public void updateMenuLinks(Host workinghost,Host updatedhost) throws DotDataExc
     }
 
 	public PaginatedArrayList<Host> search(String filter, boolean showArchived, boolean showSystemHost, int limit, int offset, User user, boolean respectFrontendRoles) throws DotDataException, DotSecurityException {
-		try {
-			Structure st = CacheLocator.getContentTypeCache().getStructureByVelocityVarName("Host");
-			String condition="";
-			
-			if(showArchived){
-				condition=" +deleted:true";
-			}else {
-				condition=" +deleted:false";
-			}
-			
-			if(UtilMethods.isSet(filter)){
-				condition += " +Host.hostName:"+filter.trim()+"*";
-			}
-			if(!showSystemHost){
-				condition += " +Host.isSystemHost:false";
-			}
-			PaginatedArrayList<Contentlet> list = (PaginatedArrayList<Contentlet>)APILocator.getContentletAPI().search("+structureInode:" + st.getInode() + condition, limit, offset, "Host.hostName", user, respectFrontendRoles);
-			
-			return convertToHostPaginatedArrayList(list);
-		} catch (Exception e) {
-			Logger.error(HostAPIImpl.class, e.getMessage(), e);
-			throw new DotRuntimeException(e.getMessage(), e);
-		}
-	}
-	
+        try {
+            Structure st = CacheLocator.getContentTypeCache().getStructureByVelocityVarName("Host");
+            String condition="";
+
+            if(showArchived){
```

Comment:
you can reduce it to condition = (showArchived)?" +deleted:true":" +deleted:false";